tests/performance/test_performance.py

```diff
@@ -0,0 +1,779 @@
+import json
+import logging
+import random
+from dataclasses import dataclass
+from datetime import timedelta
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import DatabricksError
+from databricks.sdk.retries import retried
+from databricks.sdk.service import iam, sql
+from databricks.sdk.service.iam import Group, PermissionLevel
+from databricks.sdk.service.settings import TokenAccessControlRequest, TokenPermissionLevel
+from databricks.sdk.service.workspace import (
+    AclPermission,
+    WorkspaceObjectAccessControlRequest,
+    WorkspaceObjectPermissionLevel,
+)
+
+from databricks.labs.ucx.config import WorkspaceConfig
+from databricks.labs.ucx.install import WorkspaceInstaller
+from databricks.labs.ucx.mixins.hardening import rate_limited
+from databricks.labs.ucx.workspace_access.groups import MigratedGroup
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ObjectPermission:
+    group: str
+    object_type: str
+    object_id: str
+    permission: str
+
+
+@dataclass
+class PersistedGroup:
+    group: str
+    object_type: str
+    object_id: str
+    permission: str
+
+
+verificationErrors = []
+
+def test_performance(
+    ws,
+    make_cluster_policy,
+    make_ucx_group,
+    make_warehouse,
+    make_cluster,
+    make_instance_pool,
+    make_secret_scope,
+    make_model,
+    make_experiment,
+    make_job,
+    make_pipeline,
+    make_schema,
+    sql_backend,
+    make_table,
+    make_user,
+    make_query,
+    make_alert,
+    make_dashboard,
+    make_directory,
+    make_repo,
+    make_notebook,
+    make_random,
+    env_or_skip,
+):
+    NB_OF_TEST_WS_OBJECTS = 100
+    NB_OF_FILES = 10000
+    NB_OF_TEST_GROUPS = 1000
+    NB_OF_SCHEMAS = 1
+
+    test_database = make_schema()
+    groups = create_groups(NB_OF_TEST_GROUPS, make_ucx_group, sql_backend, test_database)
+
+    users = create_users(make_user)
+
+    create_scopes(NB_OF_TEST_WS_OBJECTS, groups, make_secret_scope, sql_backend, test_database, ws)
+
```

Comment:
nit: remove blank lines between these

Reply:
make fmt will take care of these once PR is ready :)